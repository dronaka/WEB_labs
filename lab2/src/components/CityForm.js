import React from "react";
import "./styles/CityForm.css";

export default function CityForm(props) {
  return (
    <form class="add-city" onSubmit={(e) => props.onSubmit(e)}>
      <input class="input" type="text" name="cityName" placeholder="City name" required={true} />
      <input class="button" type="submit" value="Add to city"/>
    </form>
  );
}