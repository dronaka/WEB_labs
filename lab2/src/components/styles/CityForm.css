.add-city {
    display: inline;
  }
  
  .add-city > .input {
    display: inline;
  }