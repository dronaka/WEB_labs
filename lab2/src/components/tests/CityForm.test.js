import React from "react";
import renderer from "react-test-renderer";

import CityForm from "../CityForm";


describe("<CityForm /> render", () => {

  test("should has form with text input and button", () => {
    const tree = renderer
      .create(<CityForm />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
