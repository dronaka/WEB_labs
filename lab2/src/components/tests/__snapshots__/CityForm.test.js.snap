// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityForm /> render should has form with text input and button 1`] = `
<form
  class="add-city"
  onSubmit={[Function]}
>
  <input
    class="input"
    name="cityName"
    placeholder="City name"
    required={true}
    type="text"
  />
  <input
    class="button"
    type="submit"
    value="Add to city"
  />
</form>
`;
